src/formatDate.js

```diff
@@ -8,7 +8,39 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
-}
+  const separatorFrom = fromFormat[fromFormat.length - 1];
+  const dateParts = date.split(separatorFrom);
+
+  const dateObj = {};
+
+  fromFormat.slice(0, -1).forEach((part, index) => {
+    dateObj[part] = dateParts[index];
+  });
+
+  const convertYear = (year, format) => {
+    if (format === 'YYYY') {
+      if (year.length === 2) {
+        return parseInt(year) < 30 ? '20' + year : '19' + year;
+      } else {
+        return year;
+      }
+    } else if (format === 'YY') {
+      return year.slice(-2);
+    }
+
+    return year;
+  };
```

Comment:
This part of the code is a bit confusing. It would be better to split this function into two separate ones: one for converting from 'YYYY' to 'YY' and another for converting from 'YY' to 'YYYY'. This would make the code clearer and easier to read.